### ISSUE TITLE
Scrub filter for `Data,single`

### ISSUE THREAD

**Issue by opk12** [kdave/btrfs-progs#906]
I have a new USB 2 drive, attached to my desktop, with `Data,single` of 70 GB and `Metadata,DUP` of 3 GB. I have backups.

I assume that the userspace will always see I/O error and never see corrupted data, in case of a bad block. So not scrubbing means that I will sometimes need to restore, but it does not mean bad data.

* I'd like to scrub and auto-repair metadata, to prevent a slow, big restore, due to a directory corruption that breaks part of the filesystem.
* I'd not scrub 70 GB over USB 2, it's a new disk, I can quickly restore one file, and Data is unrepairable.
* I do not read the scrub log, it's just a desktop. I'll replace the disk at 10-15 years of age, no matter the bad blocks.

To sum up, I'd like to do less I/O and maintainance on my desktop, but auto-repair important things. What about a scrub filter to skip file contents or unrepairable block groups?
